### PR TITLE
Run BK exhaustive pipeline when code is pushed

### PR DESCRIPTION
**Pull request by dliappis**
## Release notes
[rn:skip]

## What does this PR do?

As we are close to sunsetting Jenkins by completing the exhaustive tests Buildkite pipeline, it makes sense to start already running the existing phases (testing and compatibility).

This commit defines the trigger to be code events, i.e. direct pushes and merge commits, similar to what we currently have in Jenkins.

## Why is it important/What is the impact to the user?

It allows us to gradually remove the corresponding phases from Jenkins to reduce CI duplication.
This will be especially useful when merging https://github.com/elastic/logstash/pull/15696 as the change is incompatible with Jenkins.

## How to test this PR locally

An example build triggered manually can be found in https://buildkite.com/elastic/logstash-exhaustive-tests-pipeline/builds/49#_
Unfortunately Windows tests will fail due to https://github.com/elastic/logstash/issues/15562

## Related issues

- Relates https://github.com/elastic/ingest-dev/issues/1722

### CODE REVIEW

File: catalog-info.yaml [hunk 1]
```diff
@@ -436,11 +436,15 @@ spec:
       repository: elastic/logstash
       pipeline_file: ".buildkite/exhaustive_tests_pipeline.yml"
       provider_settings:
-        trigger_mode: none
-      cancel_intermediate_builds: true
-      skip_intermediate_builds: true
+        build_branches: true
```
Comment (dliappis) [inline catalog-info.yaml:439]: this means that in addition to merge commits, the pipeline will be triggered when we push a new branch, e.g. `8.14`. I don't know if this is something we want, it felt right to me.

File: catalog-info.yaml [hunk 2]
```diff
@@ -436,11 +436,15 @@ spec:
...
+        build_pull_request_forks: false
+        build_pull_requests: false
+        build_tags: false
```
Comment (dliappis) [inline catalog-info.yaml:442]: conversely, I felt that we don't need to trigger this pipeline when we push tags.

File: catalog-info.yaml [hunk 3]
```diff
@@ -436,11 +436,15 @@ spec:
...
+        trigger_mode: code
+      cancel_intermediate_builds: false
+      skip_intermediate_builds: false
```
Comment (dliappis) [inline catalog-info.yaml:445]: I'd like to discuss what we want to do here; the above two lines `cancel_intermediate_builds` / `skip_intermediate_builds`, when set to `true` cancel currently running jobs (per branch) when a new commit is pushed.

This makes sense e.g. for PRs, but for intake style pipelines like this one, I think we want to let each job run completely, even if a subsequent commit lands soon after.

We can also tune this behavior per branch; e.g. [elastic-agent](https://github.com/elastic/elastic-agent/blob/main/catalog-info.yaml#L51) sets to `true` by default, but only for non `main` branches:

```
      cancel_intermediate_builds: true
      cancel_intermediate_builds_branch_filter: "!main"
      skip_intermediate_builds: true
      skip_intermediate_builds_branch_filter: "!main"
```

**Comment by dliappis** [comment]
@elasticmachine run elasticsearch-ci/docs

File: catalog-info.yaml (hunk 2 above)
Comment (jsvd) [reply catalog-info.yaml:442]: Agreed, not for tags.

File: catalog-info.yaml (hunk 1 above)
Comment (jsvd) [reply catalog-info.yaml:439]: Agreed, as long as we exclude `docs` changes (if possible)

File: catalog-info.yaml (hunk 3 above)
Comment (jsvd) [reply catalog-info.yaml:445]: it's fine that we don't cancel or skip for non-PRs, rarely we have a lot of commits landing on a given branch, so we can afford to have each commit properly/individually tested.

Comment (dliappis) [reply catalog-info.yaml:445]: Ok I'll keep things as they are then, thanks.

**Comment by dliappis** [comment]
@jsvd I addressed your comment (about skipping on docs changes). This is an example (manually triggered) build: https://buildkite.com/elastic/logstash-exhaustive-tests-pipeline/builds/56

Regarding the other comments we aligned and no further changes are needed. Please take another look when you can.

**Comment by dliappis** [comment]
@logstashmachine backport 8.12

**Pull request merged**